Use codex-linux-sandbox in unified exec

pakrym-oai opened the pull request:
Unified exec isn't working on Linux because we don't provide the correct arg0.

The library we use for pty management doesn't allow setting arg0 separately from executable. Use the same aliasing strategy we use for `apply_patch` for `codex-linux-sandbox`.

Use `#[ctor]` hack to dispatch codex-linux-sandbox calls.


Addresses https://github.com/openai/codex/issues/6450

pakrym-oai committed:
Alias codex-linux-sandbox

pakrym-oai committed:
pty: reformat tokio dependency for improved readability in Cargo.toml

pakrym-oai committed:
codex-rs: remove unused tracing dependency from lockfile

bolinfest commented on codex-rs/arg0/src/lib.rs:
```diff
@@ -144,11 +144,15 @@ where
+    for filename in &[
+        APPLY_PATCH_ARG0,
+        MISSPELLED_APPLY_PATCH_ARG0,
+        LINUX_SANDBOX_ARG0,
```
It would be good to skip `LINUX_SANDBOX_ARG0` for non-Linux.

pakrym-oai committed:
core/tests: document that test binary aliasing logic does not work on ARM

Add comments noting that the ctor logic for test binary aliasing is not supported on ARM architecture, and conditionally skip affected test on ARM targets.

bolinfest commented on codex-rs/utils/pty/src/lib.rs:
```diff
@@ -124,7 +125,7 @@ pub async fn spawn_pty_process(
     })?;
 
-    let mut command_builder = CommandBuilder::new(program);
+    let mut command_builder = CommandBuilder::new(arg0.as_ref().unwrap_or(&program.to_string()));
```
The PR body says:

> The library we use for pty management doesn't allow setting arg0 separately from executable.

so this doesn't fix it?

pakrym-oai replied:
This overrides both the arg0 and the invoked executable. This is why I need to start symlinking `codex-linux-sandbox` where we didn't need it before.

bolinfest commented on codex-rs/core/tests/suite/mod.rs:
```diff
@@ -1,4 +1,30 @@
+    let mut should_exit = true;
+
+    #[allow(clippy::unwrap_used)]
+    arg0_dispatch_or_else(|_| async {
```
`async` starts the tokio runtime, which I do not think we should be doing pre-main

pakrym-oai committed:
refactor(arg0): extract PATH entry logic into arg0_dispatch

Move path entry and alias dispatch logic from arg0_dispatch_or_else to a new arg0_dispatch function, allowing reuse in test setup. Update tests to use arg0_dispatch directly. Fix minor regex in unified_exec test. Move 'ctor' dev-dependency to dev-dependencies only in core crate.

bolinfest commented on codex-rs/arg0/src/lib.rs:
```diff
@@ -79,15 +54,45 @@ where
+    F: FnOnce(Option<PathBuf>) -> Fut,
+    Fut: Future<Output = anyhow::Result<()>>,
+{
+    let _path_entry = arg0_dispatch();
```
You lost the comment on this.

pakrym-oai committed:
arg0: move TempDir retention comment to correct function
core: fix codex-windows-sandbox dependency name
pty: add newline after env insertion for clarity
